### PR TITLE
Implement ColumnList support for runtime UDTFs

### PR DESCRIPTION
**Pull request by pearu**
As in title.

Requires https://github.com/omnisci/omniscidb-internal/pull/5274

### CODE REVIEW

File: rbc/tests/test_omnisci_column_list.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    @omnisci('int32 columns_sum1(Cursor<Column<int64>, ColumnList<T>>,'
+             ' RowMultiplier, OutputColumn<T>)',
+             T=scalar_types)
+    def columns_sum1(rowid, lst, m, out):
+        for i in range(lst.size):
+            out[i] *= 0
+            for j in range(lst.length):
+                out[i] += lst.ptrs[j][i]
+        return lst.size
```
Comment (guilhermeleobas) [inline rbc/tests/test_omnisci_column_list.py:29]: At least for me, `lst.size` and `lst.length` are a bit confusing. Does the following are equivalent?

```python
len(ColumnList) == ColumnList.length
len(ColumnList[i]) == ColumnList.size
```

File: rbc/tests/test_omnisci_column_list.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+
+@pytest.mark.parametrize("variant", ['1', "2"])
+@pytest.mark.parametrize("T", scalar_types)
+def test_columns_sum(omnisci, T, variant):
+    v = table_columns_map[T]
+    c = sql_type_map[T]
+
+    query = f'select rowid, ({v} + {v} + 1) from {omnisci.table_name} ORDER BY rowid'
```
Comment (guilhermeleobas) [inline rbc/tests/test_omnisci_column_list.py:52]: Should this query contain a `ORDER BY rowid LIMIT 1000` as well?

**Review by guilhermeleobas — APPROVED** [review]
Hi Pearu, code looks good to me. Just some minor comments

File: rbc/tests/test_omnisci_column_list.py (hunk 2 above)
Comment (pearu) [reply rbc/tests/test_omnisci_column_list.py:52]: Not really as the number of rows in the test tables is only 5 and using `LIMIT 1000` is required only to workaround omniscidb bug (without it the server would crash, the reason is unknown atm).

File: rbc/tests/test_omnisci_column_list.py (hunk 1 above)
Comment (guilhermeleobas) [reply rbc/tests/test_omnisci_column_list.py:29]: Sounds good!

**Comment by guilhermeleobas** [comment]
Hi Pearu, can you write some docs as well explaining how to use and internal attributes about ColumnList? You can put those under `doc/` folder. Don't worry if the docs don't build as expected, I can fix them in another PR.

File: rbc/tests/test_omnisci_column_list.py (hunk 1 above)
Comment (pearu) [reply rbc/tests/test_omnisci_column_list.py:29]: Resolving via introduction of ColumnList methods `nrows()` and `ncols()`.

**Pull request merged**